vector/v.info/parse.c

```diff
@@ -87,3 +87,36 @@ void parse_args(int argc, char **argv, char **input, char **field, int *history,
         }
     }
 }
+
+void parse_history_json(char *buf, char *command, char *gisdbase,
+                        char *location, char *mapset, char *user, char *date,
+                        char *mapset_path, JSON_Array *record_array)
+{
+
+    if (strncmp(buf, "COMMAND:", 8) == 0) {
+        sscanf(buf, "COMMAND: %[^\n]", command);
+    }
+    else if (strncmp(buf, "GISDBASE:", 9) == 0) {
+        sscanf(buf, "GISDBASE: %[^\n]", gisdbase);
+    }
+    else if (strncmp(buf, "LOCATION:", 9) == 0) {
+        sscanf(buf, "LOCATION: %s MAPSET: %s USER: %s DATE: %[^\n]", location,
+               mapset, user, date);
+
+        JSON_Value *info_value = json_value_init_object();
+        if (info_value == NULL) {
+            G_fatal_error(
+                _("Failed to initialize JSON object. Out of memory?"));
+        }
+        JSON_Object *info_object = json_object(info_value);
```

Comment:
This assumes the order in the source file and in the functions calls. Perhaps safe assumption, but confusing in the context.